### ISSUE TITLE
Columns are typed "REAL" if they are integers with some NaN/blanks

### ISSUE THREAD

**Issue by simonw** [simonw/csvs-to-sqlite#5]
This is bad. If a column has all integers and some blanks it should result in an INTEGER.

Example: this CSV https://github.com/openelections/openelections-data-ca/blob/master/2016/20161108__ca__general__yolo__precinct.csv produces this SQL table:

    CREATE TABLE "2016/20161108__ca__general__yolo__precinct" (
    "county" TEXT,
      "precinct" INTEGER,
      "office" INTEGER,
      "district" REAL,
      "party" REAL,
      "candidate" INTEGER,
      "votes" INTEGER
    ,
    FOREIGN KEY (county) REFERENCES county(id),
        FOREIGN KEY (party) REFERENCES party(id),
        FOREIGN KEY (precinct) REFERENCES precinct(id),
        FOREIGN KEY (office) REFERENCES office(id),
        FOREIGN KEY (candidate) REFERENCES candidate(id))

**Comment by simonw** [comment]
https://stackoverflow.com/questions/34383000/pandas-to-sql-all-columns-as-nvarchar/34384664 may be useful here

**Comment by rgieseke** [comment]
I ran into the same problem, in my https://github.com/rgieseke/pandas-datapackage-reader reader I convert columns with missing integers to dtype `object`, but that doesn't help with writing to sql. 

The Tableschema library (https://github.com/frictionlessdata/tableschema-py) might be helpful here for inferring data types:

test.csv
```
a,b
1,0.1
2,0.2
,0.3
4,
```

```py
from tableschema import infer

infer("test.csv")
```

```
{
  'fields': [
    {'format': 'default', 'name': 'a', 'type': 'integer'},
    {'format': 'default', 'name': 'b', 'type': 'number'}
  ],
   'missingValues': ['']
}
```

**Comment by simonw** [comment]
Fixed in a8ab5248f4a071bf2bfdb545a56a2cc27ef9be59 and 0997b7bef292cd9dfd9529d2d48d294d936f3fca

**Comment by simonw** [comment]
@rgieseke your pandas-datapackage-reader tool looks fascinating, I'll definitely have a play with that.
